### PR TITLE
Fix #1894 Cursor position fixed in notes

### LINKED ISSUE #1894: Text cursor lies at the beginning of text in Notes
**Describe the bug**

If there are any notes saved, and the user opens the notes again, then the cursor is found at the beginning of the text instead of at the end.
 
**Expected behavior**

The cursor should be at the end of the text.

**Steps to reproduce the behavior:**
1. Save some notes.
2. Open notes again.
3. Observe the cursor

**Screenshots**

[Link](https://imgur.com/dRNmnXU)

**Environment**
- Version of Kiwix Android : 3.2.1
- Device : Redmi 4
- OS version : Nougat

### PR DESCRIPTION
**Pull request by sonusourav**
Fixes #1894 

## Changes
Brought cursor to the end of saved notes

## Screenshot

[Link](https://imgur.com/Ut7g4Cg)

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java [hunk 1]
```diff
@@ -376,6 +376,7 @@ private void displayNote() {
       }
 
       addNoteEditText.setText(contents.toString()); // Display the note content
+      addNoteEditText.setSelection(addNoteEditText.getText().length()-1);
```
Comment (abdulwd) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java]:
```suggestion
      addNoteEditText.setSelection(addNoteEditText.getText().length() - 1);
```

Comment (sonusourav) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java]: Done

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java [hunk 2]
```diff
@@ -81,23 +81,19 @@
...
+  @Inject SharedPreferenceUtil sharedPreferenceUtil;
+  @Inject ZimReaderContainer zimReaderContainer;
   private Unbinder unbinder;
-
   private String zimFileName;
   private String zimFileTitle;
   private String articleTitle;
   // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
   private String articleNotefileName;
+  // Keeps track of state of the note (whether edited since last save)
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java]: this is in the wrong location now, please revert all the changes except
`addNoteEditText.setSelection(addNoteEditText.getText().length() - 1);`

Comment (sonusourav) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java]: Done

**Pull request merged**